core/agents/orchestrator.py

```diff
@@ -163,11 +164,23 @@ def create_agent(self, prev_response: Optional[AgentResponse]) -> BaseAgent:
             if prev_response.type == ResponseType.INPUT_REQUIRED:
                 # FIXME: HumanInput should be on the whole time and intercept chat/interrupt
                 return HumanInput(self.state_manager, self.ui, prev_response=prev_response)
-            if prev_response.type == ResponseType.UPDATE_EPIC:
-                return TechLead(self.state_manager, self.ui, prev_response=prev_response)
             if prev_response.type == ResponseType.TASK_REVIEW_FEEDBACK:
                 return Developer(self.state_manager, self.ui, prev_response=prev_response)
 
+        status = state.current_task.get("status") if state.current_task else None
```

Comment:
Shouldn't this go down between "Ask the developer to break down current task" and "Execute next step in the task" ? 

Also nitpick: I like the `current_task_status` (that you used in TechLead) name more, since it's more descriptive.

Reply:
I think i started implementation by placing it there but then moved it above. Don't remember why now, I'll take a look again.